gobblin-restli/gobblin-flow-config-service/gobblin-flow-config-service-server/src/main/java/org/apache/gobblin/service/modules/restli/FlowConfigsV2ResourceHandler.java

```diff
@@ -248,8 +251,22 @@ public CreateKVResponse<ComplexResourceKey<FlowId, FlowStatusId>, FlowConfig> cr
     // Return conflict and take no action if flowSpec has already been created
     if (this.flowCatalog.exists(flowSpec.getUri())) {
       log.warn("FlowSpec with URI {} already exists, no action will be taken", flowSpec.getUri());
-      return new CreateKVResponse<>(new RestLiServiceException(HttpStatus.S_409_CONFLICT,
-          "FlowSpec with URI " + flowSpec.getUri() + " already exists, no action will be taken"));
+      try {
+        FlowSpec storedFlowSpec = this.flowCatalog.getSpecs(flowSpec.getUri());
+        if (!storedFlowSpec.isScheduled()) {
+          log.warn("FlowSpec Already Exists As Adhoc Flow  with URI: " + flowSpec.getUri());
+          if (!flowSpec.isScheduled()) {
+            flowSpecExistsForAdhocFlow.mark();
+          }
+        } else {
+          log.warn("FlowSpec Already Exists As Scheduled Flow with URI: " + flowSpec.getUri());
+        }
+      } catch (SpecNotFoundException e) {
+        log.error("Error Retrieving FLow For Existing Flow With URI: " + flowSpec.getUri());
```

Comment:
this is not required, since the if block has already checked for existence of flowSpec